src/Go/ui/GoGameboardWrapper.tsx

```diff
@@ -0,0 +1,293 @@
+    if (boardState.previousPlayer === playerColors.white) {
+      passTurn(boardState);
+      updateBoard(boardState);
+    }
+    if (boardState.previousPlayer === null) {
+      endGame();
+      return;
+    }
+
+    setTimeout(() => {
+      opponent !== opponents.none && takeAiTurn(boardState);
+    }, 100);
+  }
+
+  async function takeAiTurn(board: BoardState) {
+    if (board.previousPlayer === null) {
+      return;
+    }
+    setWaitingOnAI(true);
+    const initialState = getStateCopy(board);
+    const move = await getMove(initialState, playerColors.white, opponent);
+
+    if (move.type === playTypes.pass) {
+      SnackbarEvents.emit(`The opponent passes their turn; It is now your turn to move.`, ToastVariant.WARNING, 4000);
+      updateBoard(initialState);
+      return;
+    }
+
+    if (move.type === playTypes.gameOver || move.x === null || move.y === null) {
+      endGame();
+      return;
+    }
+
+    const updatedBoard = await makeMove(initialState, move.x, move.y, playerColors.white, false);
+
+    if (updatedBoard) {
+      setTimeout(() => {
+        Player.go.boardState = updatedBoard;
+        rerender();
+
+        // Delay captures a short amount to make them easier to see
+        setTimeout(() => {
+          const newBoard = updateCaptures(updatedBoard, playerColors.white);
+          updateBoard(newBoard);
+          setWaitingOnAI(false);
+        }, 100);
+      }, 500);
+    }
+  }
+
+  function newSubnet() {
+    setScoreOpen(false);
+    setSearchOpen(true);
+  }
+
+  function resetState(newBoardSize = boardSize, newOpponent = opponent) {
+    setScoreOpen(false);
+    setSearchOpen(false);
+    setOpponent(newOpponent);
+    if (boardState.previousPlayer !== null && boardState.history.length) {
+      resetWinstreak(boardState.ai);
+    }
+
+    const newBoardState = getNewBoardState(newBoardSize, newOpponent);
+    if (newOpponent === opponents.Illuminati) {
+      applyHandicap(newBoardState, 4);
+    }
+    updateBoard(newBoardState);
+  }
+
+  function updateBoard(initialBoardState: BoardState) {
+    const boardState = getStateCopy(initialBoardState);
+    Player.go.boardState = boardState;
+    setScore(getScore(boardState));
+    rerender();
+  }
+
+  function endGame() {
+    endGoGame(boardState);
+    const finalScore = getScore(boardState);
+    setScore(finalScore);
+    setScoreOpen(true);
+    rerender();
+  }
+
+  function getPriorMove() {
+    if (!boardState.history.length) {
+      return Player.go.boardState;
+    }
+    const priorBoard = boardState.history.slice(-1)[0];
+    const updatedState = getStateCopy(Player.go.boardState);
+    updatedState.board = priorBoard;
+    updatedState.previousPlayer =
+      boardState.previousPlayer === playerColors.black ? playerColors.white : playerColors.black;
+
+    return updatedState;
+  }
+
+  function showPreviousMove(newValue: boolean) {
+    if (boardState.history.length) {
+      setShowPriorMove(newValue);
```

Comment:
I found a bug where you're able to switch into "previous move" even if there isn't any board history, and it stays switched as you play. This seems to indicate that the optionswitch may not correctly be responding to the `checked` prop and isn't actually controlled.

Reply:
Good find, I'll tweak this